flowcraft/generator/templates/bandage.nf

```diff
@@ -1,50 +1,36 @@
-if (params.reference{{param_id}} != null) {
-    process bandage_{{pid}} {
-        {% include "post.txt" ignore missing %}
-
-        tag { sample_id }
-        publishDir "reports/assembly/bandage_{{pid}}/$sample_id"
-
-        input:
-        set sample_id, file(assembly) from {{input_channel}}
-        file reference from Channel.fromPath(params.reference{{param_id}})
-
-        output:
-        file "*.png"
-        file "*.svg"
-        {% with task_name="bandage" %}
-        {%- include "compiler_channels.txt" ignore missing -%}
-        {% endwith %}
-
-        script:
+has_gfa1_{{pid}} = binding.hasVariable('gfa1_{{pid}}')
+
+process bandage_{{pid}} {
+    {% include "post.txt" ignore missing %}
+
+    tag { sample_id }
+    publishDir "reports/assembly/bandage_{{pid}}/$sample_id"
+
+    input:
+    set sample_id, file(fasta) from {{input_channel}}
+    file gfa1 from has_gfa1_{{pid}} ? gfa1_{{pid}} : Channel.value("NA")
+    file reference from params.reference{{param_id}} ?
+        Channel.fromPath(params.reference{{param_id}}) :
+        Channel.value("NA")
+
+    output:
+    file "*.png"
+    file "*.svg"
+    {% with task_name="bandage" %}
+    {%- include "compiler_channels.txt" ignore missing -%}
+    {% endwith %}
+
+    script:
+    assembly = has_gfa1_{{pid}} ? gfa1 : fasta
```

Comment:
Great usage of ternary operators to avoid duplicating processes :+1: . I would suggest adding a simple comment here as well to inform what this logic accomplishes.

Reply:
Done.